timemachine/fe/free_energy.py

```diff
@@ -1141,19 +1137,25 @@ def format_rates(rs):
             print("Final replica permutation  :", hrex.replica_idx_by_state)
             print()
 
+    # Use the unbound potentials associated with the summed potential once to compute the u_kln
+    # Avoids repeated creation of underlying GPU potentials
+    assert isinstance(potential, custom_ops.SummedPotential)
+    unbound_impls = potential.get_potentials()
+
     def make_energy_decomposed_state(
-        results: Tuple[StoredArrays, List[NDArray], InitialState]
+        results: Tuple[StateIdx, StoredArrays, List[NDArray]]
     ) -> EnergyDecomposedState[StoredArrays]:
-        frames, boxes, initial_state = results
+        state_idx, frames, boxes = results
+        initial_state = initial_states[state_idx]
```

Comment:
Curiosity: what was the motivation for the change to pass in a `StateIdx` and use it to look up the `initial_state`, versus directly passing in the `InitialState` as before?

Reply:
Reverted to reduce diff: [da0f7f8](https://github.com/proteneer/timemachine/pull/1287/commits/da0f7f8f29d9c2576dc31bee7b49351d05e7cb28)